resolwe/flow/models/utils/duplicate.py

```diff
@@ -394,8 +395,21 @@ def bulk_duplicate(
         new_entities = copy_objects(
             entities, contributor, name_prefix, entity_processor
         )
-
-        entity_mapping = {old.pk: new for old, new in zip(entities, new_entities)}
+        # Copy the existing annotations to the new entity and create mapping
+        # between old and new entities.
+        new_annotations = []
+        entity_mapping = dict()
+        for old_entity, new_entity in zip(entities, new_entities):
+            entity_mapping[old_entity.pk] = new_entity
+            for old_annotation in old_entity.annotations.all():
+                new_annotations.append(
+                    AnnotationValue(
```

Comment:
Move this to the `AnnotationValue` class to keep all the logic in one place.

Reply:
Done. I have done the same with annotation fields (they have to be added to the new collections).